4337/test/utils/webauthn.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * This module provides a minimal shim to emulate the Web Authentication API implemented in browsers. This allows us to
+ * write tests where we create and authenticate WebAuthn credentials that are verified on-chain.
+ *
+ * This implementation is inspired by software authenticators found in the Awesome WebAuthn list [1].
+ *
+ * [1]: <https://github.com/herrjemand/awesome-webauthn#software-authenticators>
+ */
+
+import { p256 } from '@noble/curves/p256'
+import { ethers, BytesLike } from 'ethers'
+import CBOR from 'cbor'
+import { isArrayBuffer } from 'util/types'
```

Comment:
AFAIK this method is only available in Node. Ethers has a similar method which we could use. The motivation is to provide a better reference implementation for in-browser usage.

Reply:
We now don't use `Buffer` anymore :smile: acf4235e71d0535c3b0b8cddeb4bb16ec62aff7b